### PR TITLE
feat(cli, config-plugins): [1/3] patch based config-plugin

### PR DESCRIPTION
**Pull request by Kudo**
# Why

supersedes #26199

# How

this pr touches `@expo/config-plugins` to add the `finalized` type of mods
- add a `finalized` type of mod that will run after all the other mods
- the pr updates the implementation of `sortMods` that can sort mods by given priority

the patch base config-plugin is done by series prs:
- #26413
- #26414
- #26429 

# Test Plan

new `sortMods` unit tests passed

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/@expo/config-plugins/src/Plugin.types.ts [hunk 1]
```diff
@@ -112,9 +112,13 @@ export type Mod<Props = any> = ((
 export interface ModConfig {
   android?: {
     /**
-     * Dangerously make a modification before any other android mods have been run.
+     * Dangerously make a modification before any other platform mods have been run.
      */
     dangerous?: Mod<unknown>;
+    /**
+     * A finalized mod to run after all the other platform mods have been run.
```
Comment (EvanBacon) [inline packages/@expo/config-plugins/src/Plugin.types.ts:119]: Indicate that this is also a dangerous modifier

File: packages/@expo/config-plugins/src/plugins/mod-compiler.ts [hunk 2]
```diff
@@ -103,13 +107,15 @@ function getRawClone({ mods, ...config }: ExportedConfig) {
   return Object.freeze(JSON.parse(JSON.stringify(config)));
 }
 
-const orders: Record<string, string[]> = {
-  ios: [
+const precedences: Record<string, Record<string, number>> = {
+  ios: {
     // dangerous runs first
-    'dangerous',
+    dangerous: 2,
```
Comment (EvanBacon) [inline packages/@expo/config-plugins/src/plugins/mod-compiler.ts:113]: xit: I’d expect smaller numbers to sort toward the beginning of an array, i.e. like the sort method.

File: packages/@expo/config-plugins/src/Plugin.types.ts [hunk 3]
```diff
@@ -112,9 +112,13 @@ export type Mod<Props = any> = ((
...
+     */
+    finalized?: Mod<unknown>;
```
Comment (EvanBacon) [inline packages/@expo/config-plugins/src/Plugin.types.ts:121]: There should be a helper config plugin like withDangerousMod that wraps this finalized mod

**Comment by Kudo** [comment]
the sdk ci testing failed at unrelated place, mainly from prettier bump. @expo/config-plugins testing is passed.
> @expo/config-plugins checks passed

the @expo/cli failure would be fixed by #26414

**Review by EvanBacon — APPROVED** [review]

**Pull request merged**